src/dbnode/storage/bootstrap/process.go

```diff
@@ -372,3 +376,66 @@ func (b bootstrapProcess) newRunOptions() RunOptions {
 		).
 		SetInitialTopologyState(b.initialTopologyState)
 }
+
+// NewNamespaces returns a new set of bootstrappable namespaces.
+func NewNamespaces(
+	namespaces []ProcessNamespace,
+) Namespaces {
+	namespacesMap := NewNamespacesMap(NamespacesMapOptions{})
+	for _, ns := range namespaces {
+		namespacesMap.Set(ns.Metadata.ID(), Namespace{
+			Metadata:        ns.Metadata,
+			Shards:          ns.Shards,
+			DataAccumulator: ns.DataAccumulator,
+		})
+	}
+	return Namespaces{
+		Namespaces: namespacesMap,
+	}
+}
+
+// NewNamespaceResults creates a
+// namespace results map with an entry for each
+// namespace spoecified by a namespaces map.
+func NewNamespaceResults(
+	namespaces Namespaces,
+) NamespaceResults {
+	resultsMap := NewNamespaceResultsMap(NamespaceResultsMapOptions{})
+	for _, entry := range namespaces.Namespaces.Iter() {
+		key := entry.Key()
+		value := entry.Value()
+		resultsMap.Set(key, NamespaceResult{
+			Metadata:    value.Metadata,
+			Shards:      value.Shards,
+			DataResult:  result.NewDataBootstrapResult(),
+			IndexResult: result.NewIndexBootstrapResult(),
+		})
+	}
+	return NamespaceResults{
+		Results: resultsMap,
+	}
+}
+
+// MergeNamespaceResults merges two namespace results.
+func MergeNamespaceResults(a, b NamespaceResults) NamespaceResults {
```

Comment:
Does this need to do nil checking on `a` or `b`?

Reply:
They are both structs thankfully.  Let me add a comment about them mutating both (this is the contract the other MergeFoo(...) functions have.